src/main/scala/org/broadinstitute/dsde/firecloud/model/ModelJsonProtocol.scala

```diff
@@ -131,21 +140,71 @@ object ModelJsonProtocol {
         case None => None
         case _ => throw DeserializationException("unexpected json type for " + SIZE)
       }
-      LibrarySearchParams(term, from, size)
+      val fields: Map[String, Seq[String]] = data.getOrElse(SEARCH_FIELDS, None) match {
+        case some:JsObject => some.convertTo[Map[String,Seq[String]]]
+        case None => Map.empty
+      }
+      val aggs = data.getOrElse(FIELD_AGGREGATIONS, None) match {
+        case some: JsArray => some.convertTo[Seq[String]]
+        case None => Seq.empty
+        case _ => throw DeserializationException("unexpected json type")
+      }
+      val maxAggs = data.getOrElse(MAX_AGGREGATIONS, None) match {
+        case JsNumber(x) => Some(x.intValue)
+        case None => None
+        case _ => throw DeserializationException("unexpected json type")
+      }
+      LibrarySearchParams(term, fields, aggs, maxAggs, from, size)
+    }
+  }
```

Comment:
I don't know if this is actually better, but here's an option:
```
    override def read(json: JsValue): LibrarySearchParams = {
      val data = json.asJsObject.fields
      val term = data.getOrElse(SEARCH_STRING, None) match {
        case JsString(str) if str.trim == "" => None
        case JsString(str) => Some(str.trim)
        case None => None
        case _ => throw DeserializationException("unexpected json type for " + SEARCH_STRING)
      }

      val fields = convertingReader[Map[String, Seq[String]]](SEARCH_FIELDS, data, Map.empty)
      val aggs = convertingReader[Seq[String]](FIELD_AGGREGATIONS, data, Seq.empty)

      val from: Option[Int] = intReader(FROM, data)
      val size: Option[Int] = intReader(SIZE, data)
      val maxAggs = intReader(MAX_AGGREGATIONS, data)

      LibrarySearchParams(term, fields, aggs, maxAggs, from, size)
    }

    private def intReader(fieldName:String, data:Map[String,JsValue]):Option[Int] = {
      data.get(fieldName) match {
        case Some(x:JsNumber) => Some(x.value.toInt)
        case None => None
        case _ => throw DeserializationException(s"unexpected json type for $fieldName")
      }
    }
    private def convertingReader[T](fieldName:String, data:Map[String,JsValue], default:T):T = {
      data.getOrElse(fieldName, None) match {
        case j:JsValue => Try(j.convertTo[T]).toOption.getOrElse(
          throw DeserializationException(s"unexpected json type for $fieldName")
        )
        case None => default
      }
    }
  }
```

(I used `data.get` in `intReader` and `data.getOrElse` in `convertingReader` just for fun - they should probably be consistent)

In this approach, you could move `intReader` and `convertingReader` somewhere else to make the code readable.